File: Frontend/src/api/rag.ts
// ========================================
// RAG API — 知识库问答
// ========================================
// 后端：POST /knowledge-bases/{kb_id}/query
// ⚠️ MiniMax-M2.7-highspeed 默认会返回 <think>...</think> 推理块，
//    前端要剥掉再渲染，否则用户会看到 AI 的内心独白。
// ========================================
import http from './http'

export interface RagSource {
  chunk_id: string
  score: number | null
  source_file: string
  preview: string
}

export interface RagResponse {
  query: string
  answer: string
  sources: RagSource[]
}

/** 去掉 MiniMax 的 <think>...</think> 块（兼容非贪婪跨行匹配） */
function stripThink(s: string): string {
  return s.replace(/<think>[\s\S]*?<\/think>/g, '').trim()
}

/**
 * 知识库问答（POST /knowledge-bases/{kb_id}/query）
 * - top_k 不传 → 后端走默认
 * - LLM 调 MiniMax 通常 5~30s，前端 timeout 在 vite proxy 配 120000
 */
export async function askRAG(
  kbId: number,
  query: string,
  topK?: number,
): Promise<RagResponse> {
  const body: { query: string; top_k?: number } = { query }
  if (topK !== undefined) body.top_k = topK

  const { data } = await http.post<RagResponse>(
    `/knowledge-bases/${kbId}/query`,
    body,
  )

  // 剥思考块
  data.answer = stripThink(data.answer || '')
  return data
}

File: Backend/app/routers/rag.py
"""
RAG HTTP 路由：知识库问答。

V3 接口：
    POST /knowledge-bases/{kb_id}/query
        body: {"query": "...", "top_k": 5}
        response: {"answer": "...", "sources": [...], "query": "..."}
"""
from typing import Optional

from fastapi import APIRouter, Depends, Path
from pydantic import BaseModel, Field

from app import core, models, services
from app.core.database import get_db, AsyncSession
from app.services.rag_service import RAGService
from rag.pipeline import RAGResponse

router = APIRouter(
    prefix="/knowledge-bases",
    tags=["rag"]
)


class QueryRequest(BaseModel):
    """查询请求体。"""
    query: str = Field(..., min_length=1, max_length=2000, description="用户问题")
    top_k: Optional[int] = Field(default=None, ge=1, le=20, description="检索数量（默认 5）")


class SourceItem(BaseModel):
    """引用来源项。"""
    chunk_id: str
    score: float | None = None
    source_file: str
    preview: str


class QueryResponse(BaseModel):
    """查询响应。"""
    query: str
    answer: str
    sources: list[SourceItem]


@router.post(
    "/{kb_id}/query",
    response_model=QueryResponse,
    status_code=200,
)
async def query_knowledge_base(
        kb_id: int = Path(..., description="Knowledge Base ID"),
        body: QueryRequest = ...,
        current_user: models.user.User = Depends(core.oauth2.get_current_user),
        db: AsyncSession = Depends(get_db),
):
    """
    知识库问答接口。

    权限：需要知识库读权限。
    """
    service = RAGService(db)
    response = await service.query(
        kb_id=kb_id,
        user_query=body.query,
        current_user=current_user,
        top_k=body.top_k,
    )
    return response